split-webapp/split/src/pages/Login.js

```diff
@@ -69,94 +69,94 @@ class Login extends Component {
     const { failedAuth } = this.state;
     return (
       <React.Fragment key="LoginKey">
-        <CssBaseline />
-        <Grid
-          container
-          spacing={0}
-          direction="column"
-          alignItems="center"
-          justify="center"
-          style={{ minHeight: "100vh" }}
-        >
-          <Grid item xs={0}>
-            <Container fixed justifyContent="center">
-              <Container fixed>
-                <Box
-                  component="div"
-                  borderRadius={12}
-                  className="SignInContainer"
-                >
-                  <img
-                    src={mainLogo}
-                    style={{ width: "50%", marginTop: "10%" }}
-                    alt="Main logo for login"
-                  />
-                  <Box component="div" className="InnerContainer">
-                    <Typography component="h3" className="SignIn">
-                      Sign In
-                    </Typography>
-
-                    <TextField
-                      id="outlined-basic"
-                      label="Username"
-                      variant="filled"
-                      InputProps={{
-                        startAdornment: (
-                          <InputAdornment position="start">
-                            <AccountCircle />
-                          </InputAdornment>
-                        )
-                      }}
-                      onChange={this.handleOnChangeUser}
+        <form onSubmit={this.createDetails()}>
```

Comment:
Bug: It was calling `this.createDetails()` on every re-render.

```suggestion
        <form onSubmit={() => this.createDetails()}>
```

or

```
        <form onSubmit={this.createDetails}>
```

and turn `createDetails` into an arrow function member.